precompile/contract/interfaces.go

```diff
@@ -71,9 +71,7 @@ type ConfigurationBlockContext interface {
 
 type BlockContext interface {
 	ConfigurationBlockContext
-	// GetPredicateResults returns an byte array result of verifying the predicates
-	// of the given block.
-	GetPredicateResultsBytes() []byte
+	Extra() []byte // Exta returns the Extra field of the block header
```

Comment:
How about `HeaderExtra() []byte` and remove the comment? That way it's also implicitly documented at the call site.